src/server/replica.h

```diff
@@ -59,14 +59,18 @@ class Replica : ProtocolClient {
   // Returns true if initial link with master has been established or
   // false if it has failed.
   GenericError Start();
-  void StartMainReplicationFiber();
+  struct LastMasterSyncData {
+    std::string id;
+    std::vector<LSN> last_journal_LSNs;
+  };
+  void StartMainReplicationFiber(std::optional<LastMasterSyncData> data);
```

Comment:
nit - while optional indeed works - it makes everything longer 😄 
if you add into `LastMasterSyncData` a function `bool is_defined() const { return !last_journal_LSNs.empty();}`
you could avoid this indirection - (subjective and up to you).